java-frontend/src/main/java/org/sonar/java/se/checks/CustomUnclosedResourcesCheck.java

```diff
@@ -0,0 +1,453 @@
+  private List<MethodMatcher> factoryList;
+  private List<MethodMatcher> openingList;
+  private List<MethodMatcher> closingList;
+
+  @Override
+  public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
+    final PreStatementVisitor visitor = new PreStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  @Override
+  public ProgramState checkPostStatement(CheckerContext context, Tree syntaxNode) {
+    final PostStatementVisitor visitor = new PostStatementVisitor(context);
+    syntaxNode.accept(visitor);
+    return visitor.programState;
+  }
+
+  @Override
+  public void checkEndOfExecutionPath(CheckerContext context, ConstraintManager constraintManager) {
+    final List<ObjectConstraint> constraints = context.getState().getFieldConstraints(Status.OPENED);
+    for (ObjectConstraint constraint : constraints) {
+      Tree syntaxNode = constraint.syntaxNode();
+      String name = null;
+      if (syntaxNode.is(Tree.Kind.NEW_CLASS)) {
+        name = ((NewClassTree) syntaxNode).identifier().symbolType().name();
+      } else if (syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
+        name = ((MethodInvocationTree) syntaxNode).symbolType().name();
+      }
+      if (name != null) {
+        context.reportIssue(syntaxNode, this, "Close this \"" + name + "\".");
+      }
+    }
+  }
+
+  boolean isOpeningResource(MethodInvocationTree syntaxNode) {
+    for (MethodMatcher matcher : openingMethods()) {
+      if (matcher.matches(syntaxNode)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  List<ConstructorMatcher> constructorClasses() {
+    if (classConstructor == null) {
+      if (constructor.length() > 0) {
+        classConstructor = ImmutableList.of(new ConstructorMatcher(constructor));
+      } else {
+        classConstructor = ImmutableList.of();
+      }
+    }
+    return classConstructor;
+  }
+
+  List<MethodMatcher> factoryMethods() {
+    if (factoryList == null) {
+      factoryList = new ArrayList<>();
+      if (factoryMethod.length() > 0) {
+        factoryList = ImmutableList.of(new MethodMatcher(factoryMethod));
+      } else {
+        factoryList = ImmutableList.of();
+      }
+    }
+    return factoryList;
+  }
+
+  List<MethodMatcher> openingMethods() {
+    if (openingList == null) {
+      if (openingMethod.length() > 0) {
+        openingList = ImmutableList.of(new MethodMatcher(openingMethod));
+      } else {
+        openingList = ImmutableList.of();
+      }
+    }
+    return openingList;
+  }
+
+  List<MethodMatcher> closingMethods() {
+    if (closingList == null) {
+      if (closingMethod.length() > 0) {
+        closingList = ImmutableList.of(new MethodMatcher(closingMethod));
+      } else {
+        closingList = ImmutableList.of();
+      }
+    }
+    return closingList;
+  }
+
+  private static abstract class AbstractStatementVisitor extends CheckerTreeNodeVisitor {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AModifiersOrderCheck)